### PR TITLE
task(SDK-3238) - Fixes Input PT by changing flag to MUTABLE

### PR DESCRIPTION
**Pull request by Anush-Shand**
https://wizrocket.atlassian.net/browse/SDK-3634

### CODE REVIEW

File: clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt [hunk 1]
```diff
@@ -69,8 +68,14 @@ internal object PendingIntentFactory {
             launchIntent.flags =
                 Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
             var flagsLaunchPendingIntent = PendingIntent.FLAG_UPDATE_CURRENT
-            if (Build.VERSION.SDK_INT >= VERSION_CODES.S) {
-                flagsLaunchPendingIntent = flagsLaunchPendingIntent or PendingIntent.FLAG_IMMUTABLE
+            if (VERSION.SDK_INT >= VERSION_CODES.M) {
+                flagsLaunchPendingIntent = flagsLaunchPendingIntent or
+                        if (launchIntent.hasExtra(PTConstants.PT_INPUT_FEEDBACK)) {
+                            //  PendingIntents attached to actions with remote inputs must be mutable
+                            PendingIntent.FLAG_MUTABLE
```
Comment (CTLalit) [inline clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt:75]: what about the warning on play console, how will we handle that

File: clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt [hunk 2]
```diff
@@ -85,7 +90,7 @@ internal object PendingIntentFactory {
         intent.putExtra(PTConstants.PT_DISMISS_INTENT, true)
 
         var flagsLaunchPendingIntent = PendingIntent.FLAG_CANCEL_CURRENT
-        if (Build.VERSION.SDK_INT >= VERSION_CODES.S) {
+        if (VERSION.SDK_INT >= VERSION_CODES.M) {
```
Comment (CTLalit) [inline clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt:93]: Is this intentional?

File: clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt [hunk 3]
```diff
@@ -304,18 +309,16 @@ internal object PendingIntentFactory {
...
-                return if (renderer?.deepLinkList != null) {
-                    setPendingIntent(
-                        context,
-                        notificationId,
-                        extras,
-                        launchIntent,
-                        requestCode
-                    )
-                } else {
+                if (renderer.deepLinkList == null) {
```
Comment (CTLalit) [inline clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt:312]: why did we invert the condition from != to ==

**Review by CTLalit — COMMENTED** [review]
Check comments.

File: clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt (hunk 1 above)
Comment (Anush-Shand) [reply clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt:75]: That was related to implicit pending intents https://github.com/CleverTap/clevertap-android-sdk/issues/472
The launch intent above is explicit and hence shouldn't be a problem

File: clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt (hunk 2 above)
Comment (Anush-Shand) [reply clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt:93]: Yess. There is no reason to not set the flag for lower api version. Whoever added it, missed it I think
I have tested the changes as well

File: clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt (hunk 3 above)
Comment (Anush-Shand) [reply clevertap-pushtemplates/src/main/java/com/clevertap/android/pushtemplates/content/PendingIntentFactory.kt:312]: The logic inside the if condition has been inversed as well. Earlier it was a little confusing and had redundant code.

**Review by CTLalit — APPROVED** [review]

**Pull request merged**